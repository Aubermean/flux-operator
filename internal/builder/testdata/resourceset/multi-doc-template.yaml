apiVersion: fluxcd.controlplane.io/v1
kind: ResourceSet
metadata:
  name: bundles
  namespace: flux-system
spec:
  inputs:
    - bundle: addons
      decryption: false
      apps:
        - ingress-nginx
        - cert-manager
    - bundle: apps
      decryption: true
      apps:
        - frontend
        - backend
  resourcesTemplate: |
    ---
    apiVersion: source.toolkit.fluxcd.io/v1beta2
    kind: OCIRepository
    metadata:
      name: << inputs.bundle >>
      namespace: flux-system
    spec:
      interval: 10m
      url: oci://registry.example.com/<< inputs.bundle >>
      ref:
        tag: latest
    <<- range $app := inputs.apps >>
    ---
    apiVersion: kustomize.toolkit.fluxcd.io/v1
    kind: Kustomization
    metadata:
      name: << $app >>
      namespace: flux-system
    spec:
      interval: 10m
      prune: true
      <<- if inputs.decryption >>
      decryption:
        provider: sops
        secretRef:
          name: << inputs.bundle >>-sops
      <<- end >>
      sourceRef:
        kind: OCIRepository
        name: << inputs.bundle >>
      path: ./<< $app >>
    <<- end >>
