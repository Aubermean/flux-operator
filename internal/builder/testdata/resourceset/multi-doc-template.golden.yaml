apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: OCIRepository
metadata:
  name: addons
  namespace: flux-system
spec:
  interval: 10m
  ref:
    tag: latest
  url: oci://registry.example.com/addons
---
apiVersion: kustomize.toolkit.fluxcd.io/v1
kind: Kustomization
metadata:
  name: ingress-nginx
  namespace: flux-system
spec:
  interval: 10m
  path: ./ingress-nginx
  prune: true
  sourceRef:
    kind: OCIRepository
    name: addons
---
apiVersion: kustomize.toolkit.fluxcd.io/v1
kind: Kustomization
metadata:
  name: cert-manager
  namespace: flux-system
spec:
  interval: 10m
  path: ./cert-manager
  prune: true
  sourceRef:
    kind: OCIRepository
    name: addons
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: OCIRepository
metadata:
  name: apps
  namespace: flux-system
spec:
  interval: 10m
  ref:
    tag: latest
  url: oci://registry.example.com/apps
---
apiVersion: kustomize.toolkit.fluxcd.io/v1
kind: Kustomization
metadata:
  name: frontend
  namespace: flux-system
spec:
  decryption:
    provider: sops
    secretRef:
      name: apps-sops
  interval: 10m
  path: ./frontend
  prune: true
  sourceRef:
    kind: OCIRepository
    name: apps
---
apiVersion: kustomize.toolkit.fluxcd.io/v1
kind: Kustomization
metadata:
  name: backend
  namespace: flux-system
spec:
  decryption:
    provider: sops
    secretRef:
      name: apps-sops
  interval: 10m
  path: ./backend
  prune: true
  sourceRef:
    kind: OCIRepository
    name: apps
---
